experiments/ClimaEarth/components/ocean/oceananigans.jl

```diff
@@ -0,0 +1,322 @@
+    loc = map(instantiate, OC.Fields.location(oc_field))
+    grid = oc_field.grid
+    data = oc_field.data
+
+    # TODO: There has to be a better way
+    min_lat, max_lat = extrema(OC.φnodes(grid, OC.Center(), OC.Center(), OC.Center()))
+
+    map(points) do pt
+        FT = eltype(pt)
+
+        # The oceananigans grid does not cover the entire globe, so we should not
+        # interpolate outside of its latitude bounds. Instead we return 0
+        min_lat < pt.lat < max_lat || return FT(0)
+
+        fᵢ = OC.Fields.interpolate(to_node(pt), data, loc, grid)
+        convert(FT, fᵢ)::FT
+    end
+end
+
+# function map_interpolate!(cc_field, points, oc_field::OC.Field)
+#     loc = map(instantiate, OC.Fields.location(oc_field))
+#     grid = oc_field.grid
+#     data = oc_field.data
+
+#     map!(cc_field, points) do pt
+#         FT = eltype(pt)
+#         fᵢ = OC.Fields.interpolate(to_node(pt), data, loc, grid)
+#         convert(FT, fᵢ)
+#     end
+
+#     return nothing
+# end
+
+function surface_flux(f::OC.AbstractField)
+    top_bc = f.boundary_conditions.top
+    if top_bc isa OC.BoundaryCondition{<:OC.BoundaryConditions.Flux}
+        return top_bc.condition
+    else
+        return nothing
+    end
+end
+
+function Interfacer.remap(field::OC.Field, target_space)
+    return map_interpolate(CC.Fields.coordinate_field(target_space), field)
+end
+
+function Interfacer.remap(operation::OC.AbstractOperations.AbstractOperation, target_space)
+    evaluated_field = OC.Field(operation)
+    OC.compute!(evaluated_field)
+    return Interfacer.remap(evaluated_field, target_space)
+end
+
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:area_fraction}) = sim.area_fraction
+
+# TODO: Better values for this
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:roughness_buoyancy}) = Float32(5.8e-5)
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:roughness_momentum}) = Float32(5.8e-5)
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:beta}) = Float32(1)
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:surface_direct_albedo}) = Float32(0.06)
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:surface_diffuse_albedo}) = Float32(0.06)
+
+# NOTE: This is 3D, but it will be remapped to 2D
+Interfacer.get_field(sim::OceananigansSimulation, ::Val{:surface_temperature}) = 273.15 + sim.sim.model.tracers.T
+
+function FluxCalculator.update_turbulent_fluxes!(sim::OceananigansSimulation, fields)
+    # Only LatitudeLongitudeGrid are supported because otherwise we have to rotate the vectors
+
+    (; F_lh, F_sh, F_turb_ρτxz, F_turb_ρτyz, F_turb_moisture) = fields
+
+    # TODO: Starting ClimaCore > 0.14.33 we can use interpolate!
+    remapped_F_turb_ρτxz = CC.Remapping.interpolate(sim.remappers.remapper_fc, F_turb_ρτxz)
+    remapped_F_turb_ρτyz = CC.Remapping.interpolate(sim.remappers.remapper_cf, F_turb_ρτyz)
+
+    # set_from_extrinsic_vectors!((; u = oc_flux_u, v = oc_flux_v),
```

Comment:
delete?

Reply:
I left these as comments because we will need to deal with them as next step.